### PR TITLE
Using local kinesis in dev mode

### PR DESCRIPTION
**Pull request by mkuzdowicz**
## What does this change?
Give possibility to use local kinesis in dev mode, such that more then one person can work on GRID at the same time

## How can success be measured?
run grid locally and adding images will work on multiple machines at the same office

## Tested?
- [x] locally
- [x] on TEST

### CODE REVIEW

File: docs/local-kinesis.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+command you will run app with local mode, it will use local instance of kinesis provided by [localstack](https://github.com/localstack/localstack) on http://localhost:4568
+
+it is setup in `docker-compose.yml` file
+
+if you want to cleanup your local docker environment run 
+
+     docker-compose down -v --remove-orphans
+
+if you want to do a lockup what local kinesis contains run the following command:
```
Comment (jonathonherbert) [inline docs/local-kinesis.md]:
```suggestion
if you want to do a lookup what local kinesis contains run the following command:
```

File: dev-start.sh [hunk 2]
```diff
@@ -104,12 +104,30 @@ changeNodeVersion() {
   fi
 }
 
+setupLocalKinesis() {
+  sleep 3s
+  # java sdk use CBOR protocol
+  # which does not wokr wih localstack kinesis which use kinesislite
```
Comment (jonathonherbert) [inline dev-start.sh]:
```suggestion
  # which does not work with localstack kinesis -- localstack uses kinesislite
```

File: dev-start.sh [hunk 3]
```diff
@@ -104,12 +104,30 @@ changeNodeVersion() {
...
+  export AWS_CBOR_DISABLE=true
+  # dummy env variables for local kinesis
+  export AWS_ACCESS_KEY_ID=foobar
+  export AWS_SECRET_ACCESS_KEY=foobar
+  export AWS_REGION=eu-west-2
+  echo 'cretaing local kinesis streams'
```
Comment (jonathonherbert) [inline dev-start.sh]:
```suggestion
  echo 'creating local kinesis streams'
```

File: dev-start.sh [hunk 4]
```diff
@@ -104,12 +104,30 @@ changeNodeVersion() {
   fi
 }
 
+setupLocalKinesis() {
+  sleep 3s
+  # java sdk use CBOR protocol
+  # which does not work with localstack kinesis which use kinesislite
+  export AWS_CBOR_DISABLE=true
+  # dummy env variables for local kinesis
+  export AWS_ACCESS_KEY_ID=foobar
```
Comment (akash1810) [inline dev-start.sh]: could we use the `--profile` cli flag rather than env vars here?

File: dev-start.sh [hunk 5]
```diff
@@ -104,12 +104,30 @@ changeNodeVersion() {
...
+  export AWS_SECRET_ACCESS_KEY=foobar
+  export AWS_REGION=eu-west-2
```
Comment (akash1810) [inline dev-start.sh]: why a different region from the rest of the app (eu-west-1)?

File: dev-start.sh [hunk 6]
```diff
@@ -104,12 +104,30 @@ changeNodeVersion() {
...
+  echo 'creating local kinesis streams'
+  # ignore stream already exists error
+  set +e
+  stream_name='media-service-DEV-ThrallMessageQueue-1N0T2UXYNUIC9'
+  aws --endpoint-url=http://localhost:4568 kinesis create-stream --shard-count 1 --stream-name "${stream_name}"
```
Comment (akash1810) [inline dev-start.sh]: does `create-stream` behave gracefully if the stream already exists?

File: dev-start.sh (hunk 5 above)
Comment (mkuzdowicz) [reply dev-start.sh]: it does not matter it is localhost just for AWS cli this variable need to exists with anything, will change to the same for consistency

File: dev-start.sh (hunk 4 above)
Comment (mkuzdowicz) [reply dev-start.sh]: it does not matter :) those variables need to exists with whatever
This is more for AWS cli rather then for localstack on localhost so actual kinesis

File: dev-start.sh (hunk 6 above)
Comment (mkuzdowicz) [reply dev-start.sh]: `set +e` solves that then it will just fail silently (stream already exists...) and move on

File: dev-start.sh (hunk 4 above)
Comment (mkuzdowicz) [reply dev-start.sh]: done

File: dev-start.sh (hunk 5 above)
Comment (mkuzdowicz) [reply dev-start.sh]: done

**Pull request merged**